packages/base-driver/lib/basedriver/helpers.js

```diff
@@ -66,73 +67,157 @@ async function retrieveHeaders (link) {
   return {};
 }
 
-function getCachedApplicationPath (link, currentAppProps = {}) {
+function getCachedApplicationPath (link, currentAppProps = {}, cachedAppInfo = {}) {
   const refresh = () => {
     logger.debug(`A fresh copy of the application is going to be downloaded from ${link}`);
     return null;
```

Comment:
(opinion) `null` has precious few use-cases in JS, and this isn't one of them.  given the usage, `return false` or just `return` would work just as well.

Reply:
from my perspective returning null is more strict from typing perspective. returning `false` there is not obvious there from the method name and I don't really like `undefined` as the original meaning of it is `var is not defined` while `null` means `it's defined, but empty`